launch/launch/actions/execute_process.py

```diff
@@ -681,7 +686,19 @@ def __expand_substitutions(self, context):
         cmd = [perform_substitutions(context, x) for x in self.__cmd]
         name = os.path.basename(cmd[0]) if self.__name is None \
             else perform_substitutions(context, self.__name)
-        cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
+
+        # Peform filtering for prefix application
+        filter_str = perform_substitutions(context, self.__prefix_filter)
+        regex_filter = None
+        if len(filter_str):
+            regex_filter = re.compile(filter_str)
+        if regex_filter is not None and regex_filter.match(os.path.basename(cmd[0])):
+            cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
+
+        # Apply to all if no filter is provided or filter is overridden
+        if regex_filter is None or self.__override_filter:
+            cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
```

Comment:
@camm73 following above's comment, this can be simplified a bit: 
```suggestion
        should_apply_prefix = True  # by default
        if self.__prefix_filter is not None:  # no prefix given on construction
            prefix_filter = perform_substitutions(context, self.__prefix_filter)
            # Apply if filter regex matches (empty regex matches all strings)
            should_apply_prefix = re.match(prefix_filter, os.path.basename(cmd[0]))
        if should_apply_prefix:
            cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
```